code/modules/mob/living/carbon/human/human_movement.dm

```diff
@@ -94,12 +94,59 @@
 
 /mob/living/carbon/human/Move(NewLoc, direct)
 	. = ..()
-	if(shoes && .) // did we actually move?
+	if(.) // did we actually move?
 		if(!lying && !buckled)
 			if(!has_gravity(loc))
 				return
 			var/obj/item/clothing/shoes/S = shoes
-			S.step_action(src)
+			//Bloody footprints
+			var/turf/T = get_turf(src)

```

Comment:
I think your issue with bloody footprints is somewhere in here but I don't know why